calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -0,0 +1,309 @@
+                this_pod = pod
+                break
+        else:
+            raise KeyError('Pod not found: ' + self.pod_name)
+        print('Got pod data %s' % this_pod)
+        return this_pod
+
+    def _get_api_path(self, path):
+        """
+        Get the list of pods from the Kube API server.
+
+        e.g.
+        _get_api_path('pods')
+
+        :param path: The relative path to an API endpoint.
+        :return: A list of JSON API objects
+        :rtype list
+        """
+        bearer_token = self._get_api_token()
+        session = requests.Session()
+        session.headers.update({'Authorization': 'Bearer ' + bearer_token})
+        response = session.get(
+            'https://kubernetes-master:6443/api/v1beta3/' + path,
+            verify=False,
+        )
+        response_body = response.text
+        # The response body contains some metadata, and the pods themselves
+        # under the 'items' key.
+        return json.loads(response_body)['items']
+
+    def _get_api_token(self):
+        """
+        Get the kubelet Bearer token for this node, used for HTTPS auth.
+        :return: The token.
+        :rtype: str
+        """
+        with open('/var/lib/kubelet/kubernetes_auth') as f:
+            json_string = f.read()
+        print('Got kubernetes_auth: ' + json_string)
+
+        auth_data = json.loads(json_string)
+        return auth_data['BearerToken']
+
+    def _generate_rules(self):
+        """
+        Generate the Profile rules that have been specified on the Pod's ports.
+
+        We only create a Rule for a port if it has 'allowFrom' specified.
+
+        The Rule is structured to match the Calico etcd format.
+
+        :return list() rules: the rules to be added to the Profile.
+        """
+        inbound_rules = [
+            {
+                'action': 'allow',
+            },
+        ]
+
+        outbound_rules = [
+            {
+                'action': 'allow',
+            },
+        ]
+        return inbound_rules, outbound_rules
+
+    def _generate_profile_json(self, profile_name, rules):
+        """
+        Given a list of of Calico rules, generate a Calico Profile JSON blob
+        implementing those rules.
+
+        :param profile_name: The name of the Calico profile
+        :type profile_name: string
+        :param rules: A tuple of (inbound, outbound) Calico rules
+        :type rules: tuple
+        :return: A JSON blob ready to be loaded by calicoctl
+        :rtype: str
+        """
+        inbound, outbound = rules
+        profile = {
+            'id': profile_name,
+            'inbound_rules': inbound,
+            'outbound_rules': outbound,
+        }
+        profile_json = json.dumps(profile, indent=2)
+        print('Final profile "%s": %s' % (profile_name, profile_json))
+        return profile_json
+
+    def _apply_rules(self, profile_name, pod):
+        """
+        Generate a new profile with the specified rules.
```

Comment:
A bit confusing since the caller doesn't actually specify any rules - they are generated within this method.